docs/notebooks/contributing.ipynb

```diff
@@ -10,35 +10,103 @@
+    "cd obsplus\n",
+    "pip install -e .[dev]\n",
+    "```\n",
+    "\n",
+    "We use this [git branching model](https://nvie.com/posts/a-successful-git-branching-model/). **Bug fixes should be based on the branch named \"master\" and new features should be based on the branch named \"develop\"**. Master will be merged into develop (when needed) and a release will merge develop into master.\n",
+    "\n",
+    "\n",
+    "\n",
     "## General guidelines\n",
     "\n",
-    "1. We are planning to use this [git branching model](https://nvie.com/posts/a-successful-git-branching-model/) (or something similar) starting on version 0.1.0 when ObsPlus is expected to gain a bit more stability. Until then, we will just make feature commits and bug-fix commits directly to master and roll out releases as quickly as needed \n",
+    "\n",
+    "ObsPlus uses [Black](https://github.com/ambv/black) and [flake8](http://flake8.pycqa.org/en/latest/) for code linting. Before you make any commits please run black and flake8 on your branch like so:\n",
+    "\n",
+    "```bash\n",
+    "black .\n",
+    "flake8 .\n",
+    "```\n",
+    "Address any issues raised by flake8 then commit your code.\n",
+    "\n",
+    "\n",
+    "Use [numpy style docstrings](https://docs.scipy.org/doc/numpy/docs/howto_document.html). All public code (doesn't start with a _) should have a \"full\" docstring but private code (starts with a _) can have an abbreviated docstring.\n",
+    "\n",
+    "\n",
+    "ObsPlus makes extensive use of Python 3's [type hints](https://docs.python.org/3/library/typing.html). You are encouraged to annotate any public functions/methods with type hints. \n",
+    "\n",
+    "\n",
+    "Prefer `pathlib.Path` to strings when working with paths. However, when dealing with millions of files (eg ObsPlus' indexers) strings may be preferred for efficiency.\n",
+    "\n",
+    "\n",
+    "Ensure you have cleared all output of any notebooks before making a commit. This can be done with the script `clean_docs.py` found in the scripts folder.\n",
+    "\n",
+    "Follow [PEP8's guidelines for imports](https://www.python.org/dev/peps/pep-0008/#imports).\n",
     "\n",
     "\n",
-    "2. ObsPlus uses [Black](https://github.com/ambv/black) for code linting. If you install ObsPlus in developer mode it *should* install a pre-commit hook to run black.\n",
+    "### Example functions\n",
     "\n",
     "\n",
-    "3. Use [numpy style docstrings](https://numpydoc.readthedocs.io/en/latest/format.html).\n",
+    "```python\n",
+    "from typing import Optional\n",
     "\n",
+    "import pandas as pd\n",
     "\n",
-    "4. ObsPlus makes extensive use of Python 3's [type hints](https://docs.python.org/3/library/typing.html). You are encouraged to annotate any public functions/methods with type hints. \n",
     "\n",
+    "# example public function\n",
+    "def example_func(df: pd.DataFrame, to_add: Optional[Union[int, float]]) -> pd.DataFrame:\n",
```

Comment:
Do you want to include a `from typing import Optional, Union` import for this just to make that point that these need to be imported?

Reply:
Good catch, the example code should run when copied and pasted.